core/src/main/scala/org/http4s/rho/bits/RequestAST.scala

```diff
@@ -0,0 +1,19 @@
+package org.http4s
+package rho.bits
+
+object RequestAST {
+
+  sealed trait RequestRule
+
+  case class CaptureRule[T](reader: Request => ResultResponse[T]) extends RequestRule
+
+  case class IgnoreRule(rule: RequestRule) extends RequestRule
```

Comment:
`IgnoreRule` was added so that we can ignore `Unit` types that may pop out of `CaptureRule`s which just validate the request.